src/Host/Client/Impl/Host/BrokerClient.cs

```diff
@@ -210,5 +214,15 @@ internal abstract class BrokerClient : IBrokerClient {
         public virtual string HandleUrl(string url, CancellationToken ct) {
             return url;
         }
+
+        private async Task CheckMachineOnlineAsync() {
+            if (!Uri.IsFile) {
+                var ping = new Ping();
+                var reply = await ping.SendPingAsync(Uri.Host, 5000);
```

Comment:
Makes sense. I'll change it so it returns bool and then we decide which exception to throw